hedera-node/src/main/java/com/hedera/services/store/models/Account.java

```diff
@@ -70,6 +75,31 @@
 	private Id proxy;
 	private int autoAssociationMetadata;
 
+	//TODO: Add JavaDoc if approved
+	public void updateFromGrpcContract(
+			final Optional<JKey> newAdminKey,
+			final Optional<AccountID> newProxy,
+			final Optional<Duration> newAutoRenewPeriod,
+			final Optional<Timestamp> newExpirationTime,
+			final Optional<String> newMemo,
+			final boolean affectsExpiryOnly) {
+
+		validateFalse(!affectsExpiryOnly && !isMutable(), MODIFYING_IMMUTABLE_CONTRACT);
+		validateFalse(reducesExpiry(newExpirationTime), EXPIRATION_REDUCTION_NOT_ALLOWED);
+
+		newAdminKey.ifPresent(this::setKey);
+		newMemo.ifPresent(this::setMemo);
+		newProxy.ifPresent(p -> this.setProxy(Id.fromGrpcAccount(p)));
+		newExpirationTime.ifPresent(e -> this.setExpiry(e.getSeconds()));
+		newAutoRenewPeriod.ifPresent(a -> this.setAutoRenewSecs(a.getSeconds()));
+	}
+
+	boolean isMutable() {
+		return Optional.ofNullable(this.getKey()).map(key -> !key.hasContractID()).orElse(false);
```

Comment:
I am not certain but can't this be replaced for `return this.getKey().hasContractID();`

Reply:
Implementing the suggested solution fails 5 tests:

**Test results with the suggested changes:**
![image](https://user-images.githubusercontent.com/77110630/137259837-14220b08-aade-4263-b051-8a8a4e390347.png)

![image](https://user-images.githubusercontent.com/77110630/137259870-0a717051-7143-4104-8d40-501127421d3d.png)

**Test results without the suggested changes:**
![image](https://user-images.githubusercontent.com/77110630/137260088-57b58593-8828-4dc7-a40a-ebc71f581c20.png)

![image](https://user-images.githubusercontent.com/77110630/137260111-173a2e57-1c95-4aca-afba-4f094b283cc0.png)